src/geometry/LinearPropagator.hh

```diff
@@ -21,18 +21,26 @@ class LinearPropagator
     using Initializer_t = GeoStateInitializer;
 
   public:
+    //! Output results
+    struct OutputType
```

Comment:
It's good that you've named the class consistently with the celeritas style guide, but the reason I suggested `result_type` is because that type name is consistent with the C++ standard library return type for functors. Also, the `public:` here is used to separate type definitions from function declarations, so please move the type above that.